### DISCUSSION TITLE
how to get redshift warnings outputed?

### DISCUSSION THREAD

**Discussion by tooptoop4** [sqlalchemy/sqlalchemy#9124] [Usage Questions]
from jdbc i can see output from DML like this:
```
Warnings:
UNLOAD completed, 2112669 record(s) unloaded successfully.
```

but in python/sqlalchemy i don't get this message from .execute

**Comment by CaselIT** [comment]
Hi,

I don't think sqlalchemy has anything to get it directly. You may be able to obtain then directly from the driver connection you use to connect. Refer to this docs on how to get the connection and cursor: https://docs.sqlalchemy.org/en/14/faq/connections.html#faq-dbapi-connection

**Comment by zzzeek** [comment]
There's a feature in psycopg2 called "notice logging" which is described here: https://docs.sqlalchemy.org/en/14/dialects/postgresql.html#notice-logging  that might be where psycopg2 puts this stuff.     on the psycopg2 connection themselves they're on the connection in a collection called "notices": https://www.psycopg.org/docs/connection.html#connection.notices.  if you wanted to get these directly before SQLAlchemy resets this collection (no way to turn that off now) you would use the [after_cursor_execute](https://docs.sqlalchemy.org/en/14/core/events.html#sqlalchemy.events.ConnectionEvents.after_cursor_execute) event handler , where they'd be on `connection.connection.driver_connection`.

**Reply by tooptoop4** [reply]
@zzzeek i'm not using a cursor, anyway to still get the notices?

**Reply by zzzeek** [reply]
as mentioned use the [after_cursor_execute](https://docs.sqlalchemy.org/en/14/core/events.html#sqlalchemy.events.ConnectionEvents.after_cursor_execute) handler, that gives you the cursor
